src/main/java/org/betonquest/betonquest/BetonQuest.java

```diff
@@ -1547,6 +1548,10 @@ public String getVariableValue(final String packName, final String name, final S
                 log.warn(pack, "Could not resolve variable '" + name + "'.");
                 return "";
             }
+            if ((playerID == null || "".equals(playerID.trim())) && !var.isStaticness()) {
+                log.warn(pack, "Variable '" + name + "' cannot be executed without a player reference!");
```

Comment:
I think this would be a bit better
```suggestion
            if ((playerID == null || playerID.trim().isEmpty()) && !var.isStaticness()) {
                log.warn(pack, "Variable '" + name + "' cannot be executed without a player reference!");
```

Reply:
I just implemented your suggestion but now PMD says that it is inefficient to check for empty strings this way. Should I just suppress this error?